crates/ty_python_semantic/src/types.rs

```diff
@@ -284,6 +284,81 @@ fn class_lookup_cycle_initial<'db>(
     Place::bound(Type::Never).into()
 }
 
+pub(crate) trait VarianceInferable<'db>: Sized {
```

Comment:
We are (starting to) try to cut back on the size of this file. Could you move the new variance-related stuff into a new `types/variance.rs` submodule?

Reply:
Would you move all the trait impls too?